### ISSUE TITLE
Feature Request: Containers for Flutter

### ISSUE THREAD

**Issue by davidjeba** [flutter/flutter#109377]
This feature request is for a large scale marketplace like app [crude way of explanation] hopefully you may understand from the image attached

Imagine a super large marketplace app has multiple modules
1.restaurants
2.delivery
3.offers/deals

if flutter can provide native  containerization support in both firebase and flutter it would be a super huge advantage
eg:
firebase: container "restaurant"
restaurant container will have its own firestore setup

firebase: container "delivery"
delivery container will have its own firestore setup

sameway, in flutter, if there can be
container "restaurant"
connected to its own "frs.restaurantdb" firestore container
and
container "delivery"
connected to its own "frs.deliverydb" firestore container

Technically, the traffic in one container will not affect the performance of the apps in other containers
all app modules inside the fluttermaster apps are independently scalable, can be suspended, restarted, terminated, routed etc

![fluttainers](https://user-images.githubusercontent.com/70262709/184088730-684716ff-977b-451f-b91b-58a2728a9b61.png)
https://i.stack.imgur.com/xtwbh.png

it should be easily initiated like
create container "restaurantapp" which also creates a container "frs.restaurantdb" in firebase and map it
set container "restaurantapp" to "frs.restaurantdb23" to connect a frontend app to any firestore containers

**Comment by exaby73** [comment]
Hello @davidjeba. This does not seem like a valid request. Flutter is a UI Toolkit. The developer is meant to implement their business logic. Firebase already supports using multiple Firebase projects in a single app. Other than that, Flutter can be used to build large scale applications. Google Pay is a great example of it. So you can isolate your modules through good app architecture anyways.

Due to the above reasons, I am closing this issue. If you disagree, please comment and we can reopen it. Thank you

**Comment by davidjeba** [comment]
Yes i kind of knew that this maynot be looked at as a valid request.
but for an expert team creating an architecture like this is not a challenge
https://bytes.swiggy.com/architecture-and-design-principles-behind-the-swiggys-delivery-partners-app-4db1d87a048a

my request is, if flutter can support containers/dockers which are scalable apps in a master app
the solutions that we may have in the future might be simpler by approach itself

in a nutshell:
how to create a dockerized flutter app without need a complext architecture
![image](https://user-images.githubusercontent.com/70262709/184097348-43c985f5-498d-4341-acf8-3bbfab8dee52.png)
My request is specifically on how to implement the same level of complex app in a easier approach

Strength of flutter over react native is its simplicity

**Comment by exaby73** [comment]
I understand your use case. Though this feature request would go against said simplicity of Flutter. Something similar you could do is to implement multiple Flutter activities using Flutter modules and [add-to-app](https://docs.flutter.dev/development/add-to-app). Also, I do not think containers and/or Docker would work on mobile platforms. An entire new containerization technology would have to be built to support this, or existing. You can run Docker containers on [an Android phone](https://stackoverflow.com/questions/53527277/is-it-possible-to-run-containers-on-android-devices) but it requires the user to root their device to work. This is not a feasible request, therefore I am keeping it closed
